File: src/main/java/net/dean/jraw/paginators/Sorting.java
package net.dean.jraw.paginators;

/**
 * Represents how the reddit API chooses what it returns in a Paginator
 */
public enum Sorting {
    HOT,
    NEW,
    RISING,
    CONTROVERSIAL(true),
    TOP(true);

    private boolean requiresTimePeriod;

    Sorting() { this(false); }

    Sorting(boolean requiresTimePeriod) {
        this.requiresTimePeriod = requiresTimePeriod;
    }

    public boolean requiresTimePeriod() { return requiresTimePeriod; }
}

File: lib/src/main/kotlin/net/dean/jraw/models/Sorting.kt
package net.dean.jraw.models

/**
 * Represents how the reddit API chooses what it returns in a Paginator
 */
interface Sorting {

    /**
     * If this sorting method must also include a [TimePeriod]. For example, sorting by [SubredditSort.TOP]
     * is ambiguous, reddit doesn't know what posts to include when sorting. Sorting by [SubredditSort.TOP] of the past [TimePeriod.HOUR]
     * is not ambiguous.
     */
    val requiresTimePeriod: Boolean

    val name: String
}

